pkg/oci/puller/puller.go

```diff
@@ -56,14 +58,23 @@ func (p *Puller) Pull(ctx context.Context, artifactType oci.ArtifactType, ref, d
 	}
 	repo.Client = p.Client
 
+	rootDesc, _, err := repo.FetchReference(ctx, repo.Reference.Reference)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch root descriptor: %w", err)
+	}
+
+	var artifactType oci.ArtifactType
 	copyOpts := oras.CopyOptions{}
 	copyOpts.Concurrency = 1
-	if artifactType == oci.Plugin {
+	if rootDesc.MediaType == v1.MediaTypeImageIndex {
 		plt := &v1.Platform{
 			OS:           os,
 			Architecture: arch,
 		}
 		copyOpts.WithTargetPlatform(plt)
+		artifactType = oci.Plugin
```

Comment:
We should get this from the descriptor, not just assuming that if a fat manifest then it's a plugin.

Reply:
Completely agreed!